packages/aws-cdk/lib/settings.ts

```diff
@@ -40,7 +42,12 @@ export class Configuration {
     this.projectConfig = await loadAndLog(PROJECT_CONFIG);
     this.projectContext = await loadAndLog(PROJECT_CONTEXT);
 
-    this.context = new Context(this.projectConfig, [CONTEXT_KEY], this.projectContext);
+    await this.migrateLegacyContext();
+
+    this.context = new Context(
+        this.commandLineContext,
+        this.projectConfig.subSettings([CONTEXT_KEY]).makeReadOnly(),
+        this.projectContext);
```

Comment:
This is a bit tricky, can we special-case it a bit more? Feel wrong to migrate all context that has `:` in it, or we should block this from user-defined context keys.

Reply:
I'll agree it's not fantastic, but I'd prefer just to take out the migration behavior a couple of releases in the future (maybe for GA) rather than complicate things now.